### DISCUSSION TITLE
How to disable/remove OpenSSL from Syslog-ng?

### DISCUSSION THREAD

**Discussion by dinethjay** [syslog-ng/syslog-ng#4617] [Q&A]
We’re using Syslog-ng 3.31 on an embedded system. I ran into issues booting up and I’m pretty sure this is due to OpenSSL being used in syslog-ng needing entropy, similar to [this issue](https://github.com/syslog-ng/syslog-ng/issues/3124).

I’m only doing basic logging on the device so I’d like to check whether there is a way to build syslog-ng without OpenSSL. When I try building without OpenSSL I can see that openssl is a “hard dependency”, but I’d just like to check whether there is a way to do this - as I imagine many applications of syslog-ng will not require OpenSSL.

Any help is appreciated!

**Comment by MrAnno** [comment]
OpenSSL is a mandatory dependency of syslog-ng.

This comment describes how you can resolve the entropy issue at startup without recompiling syslog-ng:
https://github.com/syslog-ng/syslog-ng/issues/3124#issuecomment-610762978

If you still want to build without OpenSSL (not supported), it is possible in theory by making a few code changes:
- remove OpenSSL-related checks from `configure.ac` and flags from `Makefiles`
- follow the compiler errors and remove crypto- and TLS-related functionality:
```
lib/cfg.c:54: #include <openssl/sha.h>
lib/compat/openssl_support.c:27: #include <openssl/ssl.h>
lib/crypto.c:34: #include <openssl/rand.h>
lib/host-id.c:28: #include <openssl/rand.h>
lib/transport/tls-context.c:33: #include <openssl/x509_vfy.h>
lib/transport/tls-session.c:28: #include <openssl/x509_vfy.h>
lib/transport/tls-verifier.c:26: #include <openssl/x509v3.h>
lib/transport/transport-tls.c:29: #include <openssl/ssl.h>
lib/uuid.c:27: #include <openssl/rand.h>
modules/afsocket/afsocket-signals.h:28: #include <openssl/x509v3.h>
modules/afsql/afsql.c:40: #include <openssl/evp.h>
modules/cryptofuncs/cryptofuncs.c:31: #include <openssl/evp.h>
modules/secure-logging
tests/loggen/loggen_helper.c:34: #include <openssl/err.h>
tests/loggen/loggen_helper.h:35: #include <openssl/ssl.h>
tests/loggen/ssl_plugin/ssl_plugin.c:39: #include <openssl/crypto.h>
...
```

**Reply by dinethjay** [reply]
Thanks. Unfortunately the tips in the discussion thread you linked did not work for me. I provided the environment variable RANDFILE which I had filled with 256 bytes from /dev/urandom. Through strace I can see it being opened and read by the openssl functions in crypto_init but the subsequent getrandom calls still block, causing syslog-ng to hang. Not sure why this is happening, as from the documentation, it looks like just providing the seed file should be enough.

**Reply by dinethjay** [reply]
Hi MrAnno, when tracing I see that syslog-ng gets blocked on a call to [_create_host_id](https://github.com/syslog-ng/syslog-ng/blob/66123c955b29298ba6e68cdc77b9a0fdb4aded36/lib/host-id.c#L33). There is a RAND_bytes call which is used to generate a 32 bit secure number for the host-id.

Since we only use logging locally I'm not sure if we need this. Do you know if there is a way to avoid generating this host-id? I can see it is only generated once on startup and then the value in the persist file is reused. I couldn't find much info about host-id in the documentation.

Thanks for your help!
